src/plots/cartesian/axes.js

```diff
@@ -846,7 +847,27 @@ axes.tickIncrement = function(x, dtick, axrev, calendar) {
     var axSign = axrev ? -1 : 1;
 
     // includes linear, all dates smaller than month, and pure 10^n in log
-    if(isNumeric(dtick)) return x + axSign * dtick;
+    if(isNumeric(dtick)) {
+        // Note 1:
+        // 0.3 != 0.1 + 0.2 but 0.3 == ((10 * 0.1) + (10 * 0.2)) / 10
+        // Attempt to use integer steps to increment
+        var magic = 1 / dtick;
+        var newX = (
+            magic * x +
+            magic * axSign * dtick
+        ) / magic;
+
+        // Note 2: now we may also consider rounding to cover few more edge cases
+        var lenDt = ('' + dtick).length;
+        var lenX0 = ('' + x).length;
+        var lenX1 = ('' + newX).length;
+
+        if(lenX1 > lenX0 + lenDt) { // this is likey a rounding error!
```

Comment:
@archmoj Just a thought here: if you know the number of decimals in `dtick`, you also know that the number of decimals in the final number should never exceed it. So if `newX.split('.')[1].length > dtickDecimalLength` then you can round with a precision of `dtickDecimalLength`. I have the feeling it could fix the `55%` but I may be wrong.

Reply:
@antoinerg thanks for sharing this.
That one not only depends on the `dtick` but also on the previous tick value. For example if we start from 1000.0001 and `dtick`=1, the next tick should be at `1001.001`.

Also I should clarify that the case of `55%` is a `d3` bug which is separated from what is going to be fixed in this PR i.e. providing better raw tick values.

Reply:
> That one not only depends on the `dtick` but also on the previous tick value. For example if we start from 1000.0001 and `dtick`=1, the next tick should be at `1001.001`.

Wouldn't taking the maximal number of decimals in either `dtick` or the `start` number work?